client/deposits/list/test/index.js

```diff
@@ -122,4 +135,125 @@ describe( 'Deposits list', () => {
+				deposits: mockDeposits,
+				isLoading: false,
+			} );
+
+			const { queryByRole } = render( <DepositsList /> );
+			const button = queryByRole( 'button', { name: 'Download' } );
+
+			expect( button ).not.toBeNull();
+		} );
+
+		test( 'does not render when there are no deposits', () => {
+			useDeposits.mockReturnValue( {
+				deposits: [],
+				isLoading: false,
+			} );
+
+			const { queryByRole } = render( <DepositsList /> );
+			const button = queryByRole( 'button', { name: 'Download' } );
+
+			expect( button ).toBeNull();
+		} );
+	} );
+
+	describe( 'CSV download', () => {
+		beforeEach( () => {
+			useDeposits.mockReturnValue( {
+				deposits: mockDeposits,
+				depositsCount: 2,
+				isLoading: false,
+			} );
+			useDepositsSummary.mockReturnValue( {
+				depositsSummary: {
+					count: 2,
+					total: 5000,
+				},
+				isLoading: false,
+			} );
+		} );
+
+		afterEach( () => {
+			jest.resetAllMocks();
+		} );
+
+		afterAll( () => {
+			jest.restoreAllMocks();
+		} );
+
+		test( 'should render expected columns in CSV when the download button is clicked', () => {
+			const { getByRole } = render( <DepositsList /> );
+			getByRole( 'button', { name: 'Download' } ).click();
+
+			const expected = [
+				'"Deposit Id"',
+				'Date',
+				'Type',
+				'Amount',
+				'Status',
+				'"Bank account"',
+			];
+
+			const csvContent = downloadCSVFile.mock.calls[ 0 ][ 1 ];
+			const csvHeaderRow = csvContent.split( os.EOL )[ 0 ].split( ',' );
+			expect( csvHeaderRow ).toEqual( expected );
+		} );
+
+		test( 'should match the visible rows', () => {
+			const { getByRole, getAllByRole } = render( <DepositsList /> );
+			getByRole( 'button', { name: 'Download' } ).click();
+
+			const csvContent = downloadCSVFile.mock.calls[ 0 ][ 1 ];
+			const csvRows = csvContent.split( os.EOL );
+			const displayRows = getAllByRole( 'row' );
+
+			expect( csvRows.length ).toEqual( displayRows.length );
+
+			const csvFirstDeposit = csvRows[ 1 ].split( ',' );
+			const displayFirstDeposit = Array.from(
+				displayRows[ 1 ].querySelectorAll( 'td' )
+			).map( ( td ) => td.textContent );
+
+			// Note:
+			//
+			// 1. CSV and display indexes are off by 1 because the first field in CSV is deposit id,
+			//    which is missing in display.
+			//
+			// 2. The indexOf check in amount's expect is because the amount in CSV may not contain
+			//    trailing zeros as in the display amount.
+			//
+			expect( formatDate( csvFirstDeposit[ 1 ] ) ).toBe(
+				displayFirstDeposit[ 0 ]
+			); // date
+			expect( csvFirstDeposit[ 2 ] ).toBe(
+				displayFirstDeposit[ 1 ].toLowerCase()
+			); // type
+			expect(
+				getUnformattedAmount( displayFirstDeposit[ 2 ] ).indexOf(
+					csvFirstDeposit[ 3 ]
+				)
+			).not.toBe( -1 ); // amount
+			expect( csvFirstDeposit[ 4 ] ).toBe(
+				displayFirstDeposit[ 3 ].toLowerCase()
+			); // status
+			expect( csvFirstDeposit[ 5 ] ).toBe(
+				`"${ displayFirstDeposit[ 4 ] }"`
+			); // bank account
+		} );
```

Comment:
Instead of modifying array items of `displayFirstDeposit` to compare with array items of `csvFirstDeposit`, it may be easier by just comparing `csvFirstDeposit` and an `expected` array, which is similar to what you did in the previous test for `csvHeaderRow`?

Reply:
I had considered creating an `expected` array before settling on the current approach. The main reason I didn't go with it was that unlike in the other test the comparisons/expectations here are a little more involved. Eg. see how the amount is being handled. Besides, I wanted my `expect`s to be more explicit to give a clear view of what's being compared. It's verbose, but IMO drives the point home.